### PR TITLE
Consumer to parse sudoers files

### PR DESCRIPTION
**Pull request by ramyahasini**
* Adding a new consumer to parse sudoers files.
* Files included - /etc/sudoers and all files under /etc/sudoers.d directory.
* Implemented test cases for sudoers

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -116,8 +117,22 @@ func (c Configuration) consumers(db *pkg.AgentDB) (consumers pkg.BaseConsumers)
 			consumers = append(consumers, &pkg.BaseConsumer{AgentDB: db, ParserLoader: state})
 		}
 	}
+	if len(c.Consumers.Sudoers) > 0 {
+		//get list of files to watch
+		sudoersFiles := c.getListOfFiles(fs, "sudoers")
+		for _, sudoersFile := range sudoersFiles {
+			if !c.fileBelongsToExclusionList(sudoersFile.File) {
```
Comment (nloginov) [inline cmd/main.go]: put `fileBelongsToExclusionList` in the `getListOfFiles`

File: cmd/main.go [hunk 2]
```diff
@@ -152,10 +167,18 @@ func (c Configuration) fileBelongsToExclusionList(file string) bool {
...
-func (c Configuration) genericConsumer(fs afero.Fs) []GenericFile {
+// Gets list of files to be monitored from all files/dirs listed in the config
+func (c Configuration) getListOfFiles(fs afero.Fs, consumerType string) []FileInfo {
 	logger := c.logger()
-	var genericFiles []GenericFile
-	for _, fullPath := range c.Consumers.Generic {
+	var filesToMonitor []FileInfo
+	var pathList []string
+	switch consumerType {
+	case "sudoers":
```
Comment (nloginov) [inline cmd/main.go]: Try to get rid of `consumerType`, just pass `the c.Consumers.Sudoers` or `c.Consumers.Generic` directly

File: cmd/main.go [hunk 3]
```diff
@@ -168,46 +191,46 @@ func (c Configuration) genericConsumer(fs afero.Fs) []GenericFile {
...
+		if c.checkIgnored(PathFull, fs, consumerType) {
 			continue // skip ignored file
 		}
 		switch mode := fi.Mode(); {
 		case mode.IsDir():
-			logger.Debug().Msg("Generic is dir")
+			logger.Debug().Msg("Path is a dir")
 			err := filepath.Walk(PathFull, func(path string, info os.FileInfo, err error) error {
-				if c.checkIgnored(path, fs) {
+				if c.checkIgnored(path, fs, consumerType) {
```
Comment (nloginov) [inline cmd/main.go]: Let's try to get rid of usage of `consumerType` in that `checkIgnored`

**Review by nloginov — CHANGES_REQUESTED** [review]

File: cmd/main.go (hunk 1 above)
Comment (ramyahasini) [reply cmd/main.go]: The function is getListOfFiles that had this logic, is removed. Keeping this here, with additional changes.

File: cmd/main.go (hunk 2 above)
Comment (ramyahasini) [reply cmd/main.go]: Updated code to pass the list instead.

File: cmd/main.go (hunk 3 above)
Comment (ramyahasini) [reply cmd/main.go]: Removed checkIgnored and added logic to exclude files that are already mapped to consumers from being mapped again.

**Review by nloginov — APPROVED** [review]

**Pull request merged**